pkg/reconcile/reconcilers.go

```diff
@@ -515,6 +518,13 @@ func newDeployment(reaper *api.Reaper, cassDcService string) *appsv1.Deployment
 							SecurityContext: reaper.Spec.SchemaInitContainerConfig.SecurityContext,
 							Env:             envVars,
 							Args:            []string{"schema-migration"},
+							VolumeMounts: []corev1.VolumeMount{
```

Comment:
There are a couples issues here. First, don't we want the volume mount for both containers?

Secondly, the volume mount will shadow the directory in which it is mounted. In other words the config files that the image writes into `/etc/reaper` won't be visible. I don't think we want to do that. I assume things didn't break due to defaults being supplied via env vars in the image.

Reply:
I have been thinking about how to handle this. We might want another init container that uses the Reaper image and simply copies `/etc/reaper` to another volume. Maybe call it `reaper-base-config`. The Reaper image might need to be modified so that its entry point script copies the contents of `reaper-base-config` into the `/etc/reaper` volume mount directory. This also means that the two existing containers will need to have a volume mount for `reaper-base-config`. 

@emerkle826 do you have any thoughts on how to handle this? Seems somewhat similar to what you did with the `base-config-init` init container (see [here](https://github.com/k8ssandra/k8ssandra/blob/main/charts/k8ssandra/templates/cassandra/cassdc.yaml#L170)).

My concern is that if we don't do this then we could end up with some subtle bugs down the road.

Reply:
For config files, I found that to be the simplest approach. For the `base-config-init` stuff, the idea was that we needed some config items that are in the base cassandra image, but weren't in the config-builder scripts, so they were getting clobbered. Copying them from the base cassandra image off to a mount point that starts empty, and then is mounted later in the process was a simple way to get around that.

If we need Reaper config here that exists in the base reaper image, then we could do the same thing. Us the image in an init container with a volume mount that starts empty, copy the config from the Reaper image to this volume mount, then re-mount the volume to `/etc/reaper` later.

Reply:
👍🏻 Makes sense for us to be consistent with our approach as base-config-init.